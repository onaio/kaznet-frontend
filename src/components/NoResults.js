import React, { Component } from "react";

export default class NoResults extends Component {
  render() {
    return (
      <div className="container">
        <br />
        <div className="jumbotron bg-white">
          <p align="center">
            Sorry, we could not find any results matching "{
              this.props.searchVal
            }".
          </p>
          <p align="center">Please try again?</p>
        </div>
      </div>
    );
  }
}
