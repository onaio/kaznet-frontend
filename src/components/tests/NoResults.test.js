// test NoResults
import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";

import NoResults from "../NoResults";

describe("components/NoResults", () => {
  it("renders without crashing", () => {
    shallow(<NoResults />);
  });

  it("renders the no results message", () => {
    const wrapper = mount(<NoResults searchVal="peter pan" />);

    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });
});
