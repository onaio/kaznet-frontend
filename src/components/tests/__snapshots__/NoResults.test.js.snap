// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/NoResults renders the no results message 1`] = `
<NoResults
  searchVal="peter pan"
>
  <div
    className="container"
  >
    <br />
    <div
      className="jumbotron bg-white"
    >
      <p
        align="center"
      >
        Sorry, we couldn't find any results matching "
        peter pan
        ".
      </p>
      <p
        align="center"
      >
        Please try again?
      </p>
    </div>
  </div>
</NoResults>
`;
